first steps revamped (#221)
OctoMind-dev/mintlify-docs: first-steps.mdx

---
title: First steps
description: "Generate your first tests and run them. Find bugs in your app before your user does."
icon: "lightbulb"
---

## 1. Give us a URL

We'll ask for a URL to create test cases. The URL has to be publicly accessible. We can test both in staging and in production, as long as we can access the site.

<Frame caption="First page of the setup flow - link to your website, screenshot 07/2024">
  <img src="/images/setup/setup-1-url.png" alt="enter your url screen" />
</Frame>

## 2. Name your project

> The second page will render a screenshot of the url you provided. We chexk if it is accessible. You can name your project now. We'll propose one similar to the url you provided.

Reviewer: ```suggestion
The second page will render a screenshot of the url you provided. We check if it is accessible. You can name your project now. We'll propose one similar to the url you provided.
```

But you can choose your own.

<Frame caption="Second page of the setup flow: Name your project, screenshot 07/2024">
  <img
    src="/images/setup/setup-2-project.png"
    alt="name your project screen, screenshot 07/2024"
  />
</Frame>

<div class="mt-8" />

> ## 3. Sign up

Reviewer: Decided to not mention "confirm your email/ set password" flow?

Author: It's kinda implied, I didn't wanna overstretch the steps.

<Frame caption="Project overview page after sign-up, screenshot 07/2024">
  <img
    src="/images/setup/setup-4-overview.png"
    alt="Opening the Octomind app on project overview for the first time, screenshot 07/2024"
  />
</Frame>

Our AI agent might be still going through your website, adding a **reachability test** and checking if there is a **cookie banner** and a **required login functionality** that need to be tested.

If the AI agent finds a required login, it will ask you for test credentials. It will use them to generate and run a **login test**. It will be added as dependency for other tests. So will be the **cookie banner test** if one is found.

<Warning>
These credentials are only usable for username/password logins not for social logins. Login using your Google, Facebook, etc. account will not work.

</Warning>

<Frame caption="AI agent asks for test credentials to auto-generate and run a login test, screenshot 07/2024">
  <img
    src="/images/setup/setup-5-login-credentials.png"
    alt="AI agent asks for login test credentials, screenshot 07/2024"
  />
</Frame>

## 5. Discover more test cases

> Here comes the really cool part. You can let the AI agent to through your app and discover more tests. Click on `discover`.

Reviewer: ```suggestion
Here comes the really cool part. You can let the AI agent browse through your site to discover and generate more tests. Click on `discover`.
```

  <img
    src="/images/setup/setup-7-discovery-done.png"
    alt="AI discovery has finished, screenshot 07/2024"
  />
</Frame>

## 6. View your first AI auto-generated test cases

You should have your **first active test cases** generated. The AI agent has discovered the test cases, it auto-generated the test steps and has run them to validate whether they work.

They are `on` which means they will run once you trigger a **test report**.

<Frame caption="First active test cases discovered and auto-generated - screenshot 07/2024">
  <img
    src="/images/setup/setup-8-first-tests.png"
    alt="First active test cases AI discovered and AI auto-generated - screenshot 07/2024 "
  />
</Frame>

If the agent stumbled over a test step, you might have to lend a hand to review and edit a test step.

<Frame caption="Test cases that need review, discovered and auto-generated - screenshot 07/2024">
  <img
    src="/images/setup/setup-8-review-steps.png"
    alt="Test cases that need review, AI discovered and AI auto-generated - screenshot 07/2024 "
  />
</Frame>

## 7. Test your website for the first time

> Once you have active test cases - they are **on**, you can **run** them. Create your first test report. Click on `run all` button and then `run in the app` and see if they found a bug.

Reviewer: ```suggestion
Once you have active test cases - they are **on**, you can **run** them. Create your first test report. Click on `run all` button and then `run in app` and see if they found a bug.
```
Good that you included that info here, test report and run all being connected is sth new users have to learn for conceptually at first I feel like

<Frame caption="Run your first test report - screenshot 07/2024">
  <img
    src="/images/setup/setup-9-run-tests.png"
    alt="Run your first test report - screenshot 07/2024 "
    alt="Approve or deny test cases that octomind discovered - screenshot 02/2024 "
  />
</Frame>

## 8. See your first test results

> You have created your first test report, congratulations! If your test is red we will pinpoint you to step in the app that is broken. Find out more about [test reports](/test-reports.mdx) and [debugging your tests](/debugtopus.mdx).

Reviewer: I like the thinking, but I'm not sure if "debugging" is what we want first time users to do/ it's quite technical. Maybe just hint them to "create a new test case" as the next mission to do?

Author: interesting. I'll send a new commit.
